docs/serializers.md

```diff
@@ -377,6 +378,71 @@ Both `Color` properties are serialized as strings.
 
 <!--- TEST -->
 
+### Delegating serializers
+
+In the previous example, we represented the `Color` class as string.
+String is considered to be a primitive type, therefore we used `PrimitiveClassDescriptor` and specialized `encodeString` method. 
+Now let's see what our actions would be if we have to serialize `Color` as other non-primitive type, let's say `IntArray`.
+
+An implementation of [KSerializer] for our original `Color` class is going to perform a conversion between
+`Color` and `IntArray`, but delegate the actual serialization logic to the `IntArraySerializer`
+using [encodeSerializableValue][Encoder.encodeSerializableValue] and
+[decodeSerializableValue][Decoder.decodeSerializableValue].
+
+```kotlin
+import kotlinx.serialization.builtins.IntArraySerializer
+
+class ColorIntArraySerializer : KSerializer<Color> {
+    override val descriptor = SerialDescriptor("Color", IntArraySerializer().descriptor)
```

Comment:
Would it perhaps be a better example to have a `val delegateSerializer = IntArraySerializer()` property instead of repeated constructor calls?

Reply:
repeated how? `descriptor` is only initialised once.

Reply:
@Dominaezzz, the serializer is used in the encode/decode functions below. While using an attribute to store the delegate serializer can use more memory, this is normally not particularly significant, and limited to the lifetime of the serializer itself (`ColorIntArraySerializer` in this case).

Reply:
I second that, let's not encourage people to use our constructor-like factories all over the place, they can be quite slow (though it's not the case here)